server/src/logic/middleware/auth-validaton.middleware.ts

```diff
@@ -0,0 +1,129 @@
+import { NextFunction, Request, Response } from "express";
+import * as jwt from "jsonwebtoken";
+import { Config } from "./../../config";
+import { ApiError, ERRORS, ExtractRequestBodyResult, SocoboUser } from "./../../models/index";
+import { UserService } from "./../services/index";
+
+export class AuthValidationMiddleware {
+
+  constructor (private _userService: UserService) {}
+
+  public checkRequest (req: Request): Promise<any> {
```

Comment:
You are right. I didn't implement this by myself. I would use the modelValidationMiddleware and add the needed decorators inside SocoboUser. class-validatior has a decorator for email.

This should be located into another issue.